app/collections/collaborators.jsx

```diff
@@ -0,0 +1,375 @@
+import React from 'react';
+import apiClient from 'panoptes-client/lib/api-client';
+import UserSearch from '../components/user-search';
+import checkIfCollectionOwner from '../lib/check-if-collection-owner';
+import alert from '../lib/alert';
+
+const ID_PREFIX = 'COLLECTION_COLLABORATORS_PAGE_';
+
+const POSSIBLE_ROLES = [
+  'owner',
+  'collaborator',
+  'contributor',
+  'viewer'
+];
+
+const ROLES_INFO = {
+  owner: {
+    label: 'Owner',
+    description: 'The collection creator. There can only be one.'
+  },
+  collaborator: {
+    label: 'Collaborator',
+    description: 'Collaborators have full access to add and remove subjects from the collection.'
+  },
+  contributor: {
+    label: 'Contributor',
+    description: 'Contributors can only add subjects to the collection.'
+  },
+  viewer: {
+    label: 'Viewer',
+    description: 'Viewers can see this collection even if it\'s private.'
+  }
+};
+
+export class RoleCreator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      creating: null,
+      error: null
+    };
+
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+
+    const checkboxes = this.node.querySelectorAll('[name="role"]');
+    const users = this.userSearch.value().map(option => parseInt(option.value, 10));
+    const checkedBoxes = Object.keys(checkboxes).filter(key => checkboxes[key].checked);
```

Comment:
I think it'll be clearer what's happening here if you converted the `checkboxes` NodeList using [`Array.from`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/from?v=control) instead of implicitly doing it with `Object.keys()` so you can use `Array.prototype.filter()`.